herbert_core/applications/multifit/@mfclass/private/multifit_lsqr.m

```diff
@@ -1,681 +1,202 @@
+    %              the actual index or indicies are needed inside my_func to be
+    %              able to get to the relevant lookup table(s).
+    %
+    %   state_in    Cell array containing previously saved internal states for each
+    %              element of win. This is information that can be used to reset the
+    %              internal state (e.g. random number generators) so that calculations
+    %              can be reproduced exactly for the same input parameters in plist.
+    %               The number of elements must match the number of elements in win.
+    %               The case of an empty state i.e. isempty(state_in{i}) is the
+    %              case of no stored state. Appropriate default behaviour must be
+    %              implemented; this will be the case on the initial call from
+    %              mutlifit_lsqr.
+    %               If the internal state is not needed, then reset_state and state_in
+    %              can be ignored.
+    %
+    %   store_in    Stored information that could be used in the function evaluation,
+    %              for example lookup tables that accumulate. This should be
+    %              different from the state: the values of store should not affect
+    %              the values of the calculated function, only the speed at which the
+    %              values are calculated.
+    %               The first call from multifit_lsqr it will be set to [].
+    %               If no storage is needed, then it can be ignored.
+    %
+    %   state_out   Cell array containing internal states to be saved for a future call.
+    %               The number of elements must match the number of elements in win.
+    %               If the internal state is not needed, then state_out can be set
+    %              to cell(size(win)) - but it must be set to a cell array with
+    %              the same nmber of elements as win.
+    %
+    %   store_out   Updated stored values. Must always be returned, but can be
+    %              set to [] if not used.
+    %
+    %
+    %   Typical code fragment could be:
+    %
+    %   function [wout, state_out, store_out] = my_func (win, caller, state_in, store_in,...
+    %                                                       @fun, plist, c1, c2, ...)
+    %       :
+    %   state_out = cell(size(win));    % create output argument
+    %       :
+    %   ind = caller.ind;
+    %   for i=1:numel(ind)
+    %       iw=ind(i);                  % index of workspace into lookup tables
+    %       % Set random number generator if necessary, and save if required for later
+    %       if reset_state
+    %           if ~isempty(state_in{i})
+    %               rng(state_in{i})
+    %           end
+    %       else
+    %           state_out{i} = rng;     % capture the random number generator state
+    %       end
+    %        :
+    %   end
+
+
+    hc = hpc_config;
```

Comment:
Worth specifying at the top that this is now implemented by the calls to the par or ser version